units/commit_queue.go

```diff
@@ -177,12 +177,10 @@ func (j *commitQueueJob) Run(ctx context.Context) {
 			"project_id":         cq.ProjectID,
 			"processing_seconds": time.Since(cq.ProcessingUpdatedTime).Seconds(),
 		})
-		// if it's a SourceCommandLine, check if the patch is done, and if it is, dequeue.
+		// check if the patch is done, and if it is, dequeue.
 		// It's okay if this gets to it before the notification does, since that will
 		// check if the item is still on the queue before removing it.
-		if projectRef.CommitQueue.PatchType == commitqueue.SourceCommandLine {
-			j.TryUnstick(cq)
-		}
+		j.TryUnstick(cq)
```

Comment:
TryUnstick _is_ written for only CLI queues. Can you tweak this function to not add errors if nextItem.Issue isn't a valid patch ID ? And it can maybe consider if version is set, and check if the version is done for PR items as well.

Reply:
doh I thought I checked to make sure that this was safe for pr items but I guess not. Tweaked this to mostly not care if the processing item is a pr or diff